src/installer/managed/Microsoft.NET.HostModel/AppHost/HostWriter.cs

```diff
@@ -73,9 +73,9 @@ void UpdateResources()
                     if (ResourceUpdater.IsSupportedOS())
                     {
                         // Copy resources from managed dll to the apphost
-                        new ResourceUpdater(appHostDestinationFilePath)
-                            .AddResourcesFromPEImage(assemblyToCopyResourcesFrom)
-                            .Update();
+                        using var updater = new ResourceUpdater(appHostDestinationFilePath);
+                        updater.AddResourcesFromPEImage(assemblyToCopyResourcesFrom);
+                        updater.Update();
```

Comment:
`HostWriter.CreateAppHost` should do `RetryUtil.RetryOnIOError` instead of `RetryUtil.RetryOnWin32Error` now that this doesn't use Win32 APIs.

https://github.com/dotnet/runtime/blob/1378acb90fb0d267331647005eb4edccefa5d527/src/installer/managed/Microsoft.NET.HostModel/AppHost/HostWriter.cs#L128